scripts/verify-packaged-fork-config.sh

```diff
@@ -0,0 +1,178 @@
+MINA_GENESIS_EXE=${MINA_GENESIS_EXE:-"mina-create-genesis"}
+found_config=$(echo /var/lib/coda/config_*.json)
+PACKAGED_CONFIG_JSON=${PACKAGED_CONFIG_JSON:-$found_config}
+GENESIS_LEDGER_DIR=${GENESIS_LEDGER_DIR:-"/var/lib/coda"}
+SECONDS_PER_SLOT=${SECONDS_PER_SLOT:=180}
+FORKING_FROM_CONFIG_JSON=${FORKING_FROM_CONFIG_JSON:=/var/lib/coda/mainnet.json}
+
+export MINA_LIBP2P_PASS=''
+
+workdir=$2
+mkdir -p "$workdir"
+mkdir -p "$workdir/ledgers"
+mkdir -p "$workdir/ledgers-backup"
+mkdir -p "$workdir/keys"
+chmod 700 "$workdir/keys"
+
+if [ ! -e "$workdir/keys/p2p" ]; then
+    "$MINA_EXE" libp2p generate-keypair --privkey-path "$workdir/keys/p2p"
+fi
+
+# inspired by scripts/hardfork/create_runtime_config.sh
+function generate_ledgers_tar_from_config() {
+    config_file_input=$1
+    genesis_dir=$2
+    config_file_output=$3
+    
+    config_file_tmp=$(mktemp)
+    echo "preprocessing config file..."
+    cp "$config_file_input" "$config_file_tmp"
+    sed -i -e 's/"set_verification_key": "signature"/"set_verification_key": {"auth": "signature", "txn_version": "1"}/' "$config_file_tmp"
+    
+    echo "generating genesis ledger... (this may take a while)"
+    "$MINA_GENESIS_EXE" --config-file "$config_file_tmp" --genesis-dir "$genesis_dir" --hash-output-file "$workdir/hashes.json"
+    echo "genesis ledger generated"
+    
+    echo "updating config file..."
+    
+    GENESIS_TIMESTAMP=$(jq -r '.genesis.genesis_state_timestamp' "$PACKAGED_CONFIG_JSON")
+    
+    # Pull the original genesis timestamp from the pre-fork config file
+    ORIGINAL_GENESIS_TIMESTAMP=$(jq -r '.genesis.genesis_state_timestamp' "$FORKING_FROM_CONFIG_JSON")
+    
+    DIFFERENCE_IN_SECONDS=$(($(date -d "$GENESIS_TIMESTAMP" "+%s") - $(date -d "$ORIGINAL_GENESIS_TIMESTAMP" "+%s")))
+    DIFFERENCE_IN_SLOTS=$((DIFFERENCE_IN_SECONDS / SECONDS_PER_SLOT))
+    
+    jq "{\
+    genesis: {\
+        genesis_state_timestamp: \"$GENESIS_TIMESTAMP\"\
+    },\
+    proof: {\
+        fork: {\
+            state_hash: .proof.fork.state_hash,\
+            blockchain_length: .proof.fork.blockchain_length,\
+            global_slot_since_genesis: $DIFFERENCE_IN_SLOTS,\
+        },\
+    },\
+    ledger: {\
+        add_genesis_winner: false,\
+        hash: \$hashes[0].ledger.hash,\
+        s3_data_hash: \$hashes[0].ledger.s3_data_hash\
+    },\
+    epoch_data: {\
+        staking: {\
+            seed: .epoch_data.staking.seed,\
+            hash: \$hashes[0].epoch_data.staking.hash,\
+            s3_data_hash: \$hashes[0].epoch_data.staking.s3_data_hash\
+        },\
+        next: {\
+            seed: .epoch_data.next.seed,\
+            hash: \$hashes[0].epoch_data.next.hash,\
+            s3_data_hash: \$hashes[0].epoch_data.next.s3_data_hash\
+        }\
+    }\
+    }" -M \
+    --slurpfile hashes "$workdir/hashes.json" "$1" > "$config_file_output"
+}
+
+generate_ledgers_tar_from_config "$1" "$workdir/ledgers" "$workdir/config-substituted.json"
```

Comment:
It would be best if this script were split into 2: one that verifies the tarballs given them as input, and one that does this e2e generation and then verification process. In hardfork pipelines, we can run just the verification of the final tarball we are shipping with the release, and per-PR or per-nightly, we can run the e2e test. As mentioned above, would be best if the code for generating the configs was shared between the hardfork packaging pipeline and this script.